src/layers/tile-layer-utils/TileCache.js

```diff
@@ -49,46 +52,64 @@ export default class TileCache {
   update(viewport) {
     const {_cache, _getTileData, _maxSize, _maxZoom, _minZoom} = this;
     this._markOldTiles();
-    const tileIndices = getTileIndices(viewport, _maxZoom, _minZoom);
-    if (!tileIndices || tileIndices.length === 0) {
-      return;
-    }
-    _cache.forEach(cachedTile => {
-      if (tileIndices.some(tile => cachedTile.isOverlapped(tile))) {
-        cachedTile.isVisible = true;
+    var tileIndices = getTileIndices(viewport, _maxZoom, _minZoom);
+    if(tileIndices.length > 0) {
+      if(tileIndices[0].z != _minZoom) {
+        tileIndices.push(...this._expandIndices(tileIndices))
       }
-    });
+      tileIndices = tileIndices.filter((e) => (e.z > _minZoom && e.y >= 0 && e.x >= 0) || (e.z <= _minZoom && e.x == 0 && e.y == 0))
+      var [maxY, maxX] = [Number.NEGATIVE_INFINITY, Number.NEGATIVE_INFINITY]
+      for (var tileIndex in tileIndices) {
+        var tile = tileIndices[tileIndex]
+        if(tile.x > maxX)
+          maxX = tile.x
+        if(tile.y > maxY)
+          maxY = tile.y
+      }
+      maxX = Math.min(maxX, Math.floor(MAX_WIDTH / (256 * Math.pow(2, -1 * tileIndices[0].z))))
```

Comment:
Is it also 256 in DZI, or should this be parameterized?

Reply:
discussed offline - these will be fetched